dnf/plugins/repoquery/dnf-command-repoquery.c

```diff
@@ -0,0 +1,158 @@
+/*
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Licensed under the GNU Lesser General Public License Version 2.1
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ */
+
+#include "dnf-command-repoquery.h"
+
+struct _DnfCommandRepoquery
+{
+  PeasExtensionBase parent_instance;
+};
+
+static void dnf_command_repoquery_iface_init (DnfCommandInterface *iface);
+
+G_DEFINE_DYNAMIC_TYPE_EXTENDED (DnfCommandRepoquery,
+                                dnf_command_repoquery,
+                                PEAS_TYPE_EXTENSION_BASE,
+                                0,
+                                G_IMPLEMENT_INTERFACE (DNF_TYPE_COMMAND,
+                                                       dnf_command_repoquery_iface_init))
+
+static void
+dnf_command_repoquery_init (DnfCommandRepoquery *self)
+{
+}
+
+static void
+disable_available_repos (DnfContext *ctx)
+{
+  GPtrArray *repos = dnf_context_get_repos (ctx);
+  for (guint i = 0; i < repos->len; ++i)
+    {
+      DnfRepo * repo = g_ptr_array_index (repos, i);
+      dnf_repo_set_enabled (repo, DNF_REPO_ENABLED_NONE);
+    }
+}
+
+static gint
+gptrarr_dnf_package_cmp (gconstpointer a, gconstpointer b)
+{
+  return dnf_package_cmp(*(DnfPackage**)a, *(DnfPackage**)b);
+}
+
+static gboolean
+dnf_command_repoquery_run (DnfCommand     *cmd,
+                          int              argc,
+                          char            *argv[],
+                          GOptionContext  *opt_ctx,
+                          DnfContext      *ctx,
+                          GError         **error)
+{
+  gboolean opt_available = FALSE;
+  gboolean opt_installed = FALSE;
+  g_auto(GStrv) opt_key = NULL;
+  const GOptionEntry opts[] = {
+    { "available", '\0', G_OPTION_FLAG_NONE, G_OPTION_ARG_NONE, &opt_available, "display available packages (default)", NULL },
```

Comment:
The `--available` switch looks rather unnecessary, since it's the default? It basically has no effect.

Reply:
To display all packages "available" and "installed" user have to use both options at once: `microdnf repoquery --installed --available ...`